library/spdm_crypt_lib/libspdm_crypt_crypt.c

```diff
@@ -3864,16 +3899,213 @@ static bool internal_spdm_x509_date_time_check(IN uint8_t *from,
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_2048:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSAPSS_2048:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_3072:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSAPSS_3072:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_4096:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSAPSS_4096:
+        switch (base_hash_algo) {
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_256: {
+            uint8_t signature_algo_oid_sha256RSA[] = SIGN_ALGO_SHA256RSA_OID;
+            copy_mem(oid,signature_algo_oid_sha256RSA,oid_len);
+            break;
+        }
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_384: {
+            uint8_t signature_algo_oid_sha384RSA[] = SIGN_ALGO_SHA384RSA_OID;
+            copy_mem(oid,signature_algo_oid_sha384RSA,oid_len);
+            break;
+        }
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_512: {
+            uint8_t signature_algo_oid_sha512RSA[] = SIGN_ALGO_SHA512RSA_OID;
+            copy_mem(oid,signature_algo_oid_sha512RSA,oid_len);
+            break;
+        }
+        /*rsa sha3 oid  TBD*/
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_256:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_384:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_512:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SM3_256:
+
+        default:
+            ASSERT(false);
+            return false;
+        }
+        break;
+
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P256:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P384:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P521:
+        switch (base_hash_algo) {
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_256: {
+            uint8_t signature_algo_oid_sha256ECC[] = SIGN_ALGO_SHA256ECC_OID;
+            copy_mem(oid,signature_algo_oid_sha256ECC,oid_len);
+            break;
+        }
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_384: {
+            uint8_t signature_algo_oid_sha384ECC[] = SIGN_ALGO_SHA384ECC_OID;
+            copy_mem(oid,signature_algo_oid_sha384ECC,oid_len);
+            break;
+        }
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_512: {
+            uint8_t signature_algo_oid_sha512ECC[] = SIGN_ALGO_SHA512ECC_OID;
+            copy_mem(oid,signature_algo_oid_sha512ECC,oid_len);
+            break;
+        }
+        /*ecc sha3 oid  TBD*/
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_256:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_384:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA3_512:
+        case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SM3_256:
+
+        default:
+            ASSERT(false);
+            return false;
+        }
+        break;
+
+    /*sm2 oid  TBD*/
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_SM2_ECC_SM2_P256:
+        ASSERT(false);
```

Comment:
this is will cause current SM2 cert in EMU fail.

I think we need make it pass temporarily and add SM2 specific check later.

Reply:
Yes. Because the libspdm_get_signature_algo_OID function only is called by libspdm_verify_cert_signature_algo_OID now.
And in libspdm_verify_cert_signature_algo_OID, I have added the code before:
```
    /*work around: skip the sm2*/
    if (base_asym_algo == SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_SM2_ECC_SM2_P256) {
        return true;
    }
```
I think your mean is the libspdm_get_signature_algo_OID function may be called by other function in the future.
I will fix the issue.